@xen-orchestra/xapi/pool.mjs

```diff
@@ -0,0 +1,106 @@
+import { asyncEach } from '@vates/async-each'
+import { createLogger } from '@xen-orchestra/log'
+import { Task } from '@vates/task'
+
+import { getCurrentVmUuid } from './_XenStore.mjs'
+
+const noop = Function.prototype
+
+async function pCatch(p, code, cb) {
+  try {
+    return await p
+  } catch (error) {
+    if (error.code === code) {
+      return cb(error)
+    }
+    throw error
+  }
+}
+
+const { warn } = createLogger('xo:xapi:pool')
+
+export default class Pool {
+  async emergencyShutdown() {
+    const poolMasterRef = this.pool.master
+
+    let currentVmRef
+    try {
+      currentVmRef = await this.call('VM.get_by_uuid', await getCurrentVmUuid())
+
+      // try to move current VM on pool master
+      const hostRef = await this.call('VM.get_resident_on', currentVmRef)
```

Comment:
In case the XOA is in another pool, `currentVmRef` will be `undefined`, right?
So in this case, we don't need to wait for `VM.get_resident_on`